src/components/fields/file-upload/file-upload-manager.ts

```diff
@@ -0,0 +1,321 @@
+				(upload.type === "multipart" && rawFile.size > maxSizeInB)
+			) {
+				return {
+					errorMessage:
+						maxFileSizeRule.errorMessage ||
+						ERROR_MESSAGES.UPLOAD().MAX_FILE_SIZE(maxFileSizeRule.maxSizeInKb),
+					fileType,
+					status: EFileStatus.ERROR_SIZE,
+				};
+			}
+		}
+
+		if (addedFrom === "schema") {
+			return {
+				fileType,
+				status: EFileStatus.UPLOADED,
+			};
+		}
+
+		return {
+			fileType,
+			status: EFileStatus.UPLOAD_READY,
+		};
+	};
+
+	// =============================================================================
+	// FILE STATUS HANDLERS
+	// =============================================================================
+	const injectFile = async (fileToInject: IFile, index: number) => {
+		setFiles((prev) => {
+			const updatedFiles = [...prev];
+			updatedFiles[index] = {
+				...prev[index],
+				status: EFileStatus.INJECTING,
+			};
+			return updatedFiles;
+		});
+
+		let rawFile: File;
+		if (fileToInject.dataURL) {
+			const blob = await FileHelper.dataUrlToBlob(fileToInject.dataURL);
+			rawFile = new File([blob], fileToInject.rawFile.name);
+		} else if (fileToInject.fileUrl) {
+			const response: Blob = await new AxiosApiClient("", undefined, undefined, false, {
+				responseType: "blob",
+			}).get(fileToInject.fileUrl);
+			const fileType = await FileHelper.getType(new File([response], fileToInject.rawFile.name));
+			rawFile = new File([response], fileToInject.rawFile.name, { type: fileType.mime });
+			fileToInject.dataURL = await FileHelper.fileToDataUrl(rawFile);
+		}
+		const { errorMessage, fileType } = await readFile({ ...fileToInject, rawFile });
+		const thumbnailImageDataUrl = await generateThumbnail(fileToInject, fileType.mime);
+
+		setFiles((prev) => {
+			const updatedFiles = [...prev];
+			updatedFiles[index] = {
+				...fileToInject,
+				fileItem: {
+					errorMessage,
+					id: fileToInject.fileItem?.id || generateRandomId(),
+					name: FileHelper.deduplicateFileName(
+						files.map(({ fileItem }) => fileItem?.name),
+						index,
+						rawFile.name
+					),
+					progress: 1,
+					size: rawFile.size,
+					type: fileType.mime,
+					thumbnailImageDataUrl,
+				},
+				rawFile,
+				status: EFileStatus.UPLOADED,
+			};
+			return updatedFiles;
+		});
+	};
+
+	const parseFile = async (fileToParse: IFile, index: number) => {
+		const dataURL = await FileHelper.fileToDataUrl(fileToParse.rawFile);
+		const { errorMessage, fileType, status } = await readFile({ dataURL, ...fileToParse });
+
+		setFiles((prev) => {
+			const updatedFiles = [...prev];
+			updatedFiles[index] = {
+				...fileToParse,
+				dataURL,
+				fileItem: {
+					errorMessage,
+					id: generateRandomId(),
+					name: FileHelper.deduplicateFileName(
+						files.map(({ fileItem }) => fileItem?.name),
+						index,
+						fileToParse.rawFile.name
+					),
+					size: fileToParse.rawFile.size,
+					type: fileType.mime,
```

Comment:
this line throws an error when a plain text file is uploaded. `fileType` is undefined somehow

Reply:
it appears not to be possible to determine the file type of text-based files through magic numbers, the explanation is such files don't come with any signatures: https://github.com/sindresorhus/file-type/issues/264

in view of that, I have added a fall back to use whatever is defined in the file if the detection fails.